In [72]:
import pandas as pd
import matplotlib.pyplot as plt
import hashlib
import plotly.express as px

In [73]:
df = pd.read_csv("athlete_events.csv")
df.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

Number of countries represented in the data
with specified NOC:

In [74]:
print(df['NOC'].nunique())
print(df['NOC'].unique())

230
['CHN' 'DEN' 'NED' 'USA' 'FIN' 'NOR' 'ROU' 'EST' 'FRA' 'MAR' 'ESP' 'EGY'
 'IRI' 'BUL' 'ITA' 'CHA' 'AZE' 'SUD' 'RUS' 'ARG' 'CUB' 'BLR' 'GRE' 'CMR'
 'TUR' 'CHI' 'MEX' 'URS' 'NCA' 'HUN' 'NGR' 'ALG' 'KUW' 'BRN' 'PAK' 'IRQ'
 'UAR' 'LIB' 'QAT' 'MAS' 'GER' 'CAN' 'IRL' 'AUS' 'RSA' 'ERI' 'TAN' 'JOR'
 'TUN' 'LBA' 'BEL' 'DJI' 'PLE' 'COM' 'KAZ' 'BRU' 'IND' 'KSA' 'SYR' 'MDV'
 'ETH' 'UAE' 'YAR' 'INA' 'PHI' 'SGP' 'UZB' 'KGZ' 'TJK' 'EUN' 'JPN' 'CGO'
 'SUI' 'BRA' 'FRG' 'GDR' 'MON' 'ISR' 'URU' 'SWE' 'ISV' 'SRI' 'ARM' 'CIV'
 'KEN' 'BEN' 'UKR' 'GBR' 'GHA' 'SOM' 'LAT' 'NIG' 'MLI' 'AFG' 'POL' 'CRC'
 'PAN' 'GEO' 'SLO' 'CRO' 'GUY' 'NZL' 'POR' 'PAR' 'ANG' 'VEN' 'COL' 'BAN'
 'PER' 'ESA' 'PUR' 'UGA' 'HON' 'ECU' 'TKM' 'MRI' 'SEY' 'TCH' 'LUX' 'MTN'
 'CZE' 'SKN' 'TTO' 'DOM' 'VIN' 'JAM' 'LBR' 'SUR' 'NEP' 'MGL' 'AUT' 'PLW'
 'LTU' 'TOG' 'NAM' 'AHO' 'ISL' 'ASA' 'SAM' 'RWA' 'DMA' 'HAI' 'MLT' 'CYP'
 'GUI' 'BIZ' 'YMD' 'KOR' 'THA' 'BER' 'ANZ' 'SCG' 'SLE' 'PNG' 'YEM' 'IOA'
 'OMA' 'FIJ' 'VAN' 'MDA' 'YUG' 'BAH' 'GUA' 'SRB

The different sports represented in the data:

In [75]:
df['Sport'].unique()

array(['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating',
       'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming',
       'Badminton', 'Sailing', 'Biathlon', 'Gymnastics',
       'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting',
       'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh',
       'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo',
       'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon',
       'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball',
       'Synchronized Swimming', 'Table Tennis', 'Nordic Combined',
       'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing',
       'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon',
       'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby',
       'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo',
       'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol',
       'Croquet', 'Jeu De Paume', 'Roque', 'Al

Types of Medals:

In [76]:
df['Medal'].unique()

array([nan, 'Gold', 'Bronze', 'Silver'], dtype=object)

Statistics over age of athletes:

In [77]:
print(f"mean age: {df['Age'].mean()}")
print(f"median age: {df['Age'].median()}")
print(f"min age: {df['Age'][df['Age']>0].min()}")
print(f"max age: {df['Age'].max()}")
print(f"standard deviation: {df['Age'].std()}")

mean age: 25.556898357297374
median age: 24.0
min age: 10.0
max age: 97.0
standard deviation: 6.393560847035813


Gender distribution:

In [103]:
sex_counts = df['Sex'].value_counts().reset_index()
sex_counts.columns = ['Sex', 'Count']

fig_sex_counts = px.bar(
    sex_counts,
    x='Sex',
    y='Count',
    color='Sex',
    color_discrete_map={'M': 'blue', 'F': 'red'},
    title="Distribution of Athletes by Gender"
)

fig_sex_counts.show()

In [102]:
gender_year_season = df.groupby(['Season', 'Year', 'Sex']).size().unstack(fill_value=0).reset_index()

# Summer OS
summer_df = gender_year_season[gender_year_season['Season'] == 'Summer'].melt(
    id_vars=['Year'], value_vars=['F', 'M'], var_name='Sex', value_name='Number of athletes'
)

fig_summer = px.line(
    summer_df,
    x='Year',
    y='Number of athletes',
    color='Sex',
    markers=True,
    title='Summer OS'
)

# Winter OS
winter_df = gender_year_season[gender_year_season['Season'] == 'Winter'].melt(
    id_vars=['Year'], value_vars=['F', 'M'], var_name='Sex', value_name='Number of athletes'
)

fig_winter = px.line(
    winter_df,
    x='Year',
    y='Number of athletes',
    color='Sex',
    markers=True,
    title='Winter OS'
)

# Visa figurer
fig_summer.show()
fig_winter.show()


Top 10 countries medal distribution:

In [100]:

top10_medals = (
    df.groupby('NOC')['Medal']
    .count()
    .sort_values(ascending=False)
    .iloc[:10]
    .reset_index()
)

top10_medals.columns = ['NOC', 'Count']

fig_top10_medals = px.bar(
    top10_medals,
    x='NOC',
    y='Count',
    title="Top 10 countries by total medals",
    labels={'NOC': 'Country', 'Count': 'Number of medals'}
)

fig_top10_medals.show()

# ITALIA

Anonimization of athlete names:

In [81]:
df['Name'] = df['Name'].apply(lambda x: hashlib.sha256(x.encode()).hexdigest())

Seperating Italian statistics from the data:

In [82]:
italydf_anon = df[df['NOC'] == 'ITA']
italydf_anon.head()

ID                                               Name Sex   Age  Height  \
158   62  1c1026605c0607937a724d3d839d964064f6907b3b2e0e...   M  21.0   198.0   
197   91  d5ce4ad5e9aa7549d6069b66cfefa8df14d801fd845cc7...   M  27.0   190.0   
198   92  207f8f997895f35185acbd1e26fab5176d8b3dc6f3300f...   M  21.0   180.0   
214  103  5a52f67df03607d3096242e2779913d915ccb30d26438f...   M  25.0     NaN   
218  106  e54ebcdcbc9a487cea464c1def380df0dc1ea3d7f4ba83...   M  22.0   188.0   

     Weight   Team  NOC        Games  Year  Season            City  \
158    90.0  Italy  ITA  2016 Summer  2016  Summer  Rio de Janeiro   
197    80.0  Italy  ITA  2012 Summer  2012  Summer          London   
198    73.0  Italy  ITA  2008 Summer  2008  Summer         Beijing   
214     NaN  Italy  ITA  1936 Summer  1936  Summer          Berlin   
218    96.0  Italy  ITA  1988 Summer  1988  Summer           Seoul   

                 Sport                               Event   Medal  
158             Rowing          Rowing Men's Coxless Pairs  Bronze  
197          Athletics  Athletics Men's 110 metres Hurdles     NaN  
198           Football             Football Men's Football     NaN  
214  Modern Pentathlon  Modern Pentathlon Men's Individual  Bronze  
218             Rowing       Rowing Men's Quadruple Sculls    Gold

Italian medals per sport:

In [99]:
medals_per_sport = italydf_anon[italydf_anon['Medal'].notnull()]['Sport'].value_counts().reset_index()
medals_per_sport.columns = ['Sport', 'Count']

fig_medals_per_sport = px.bar(
    medals_per_sport,
    x='Sport',
    y='Count',
    title="Italian Medals per Sport",
    labels={'Sport': 'Sport', 'Count': 'Number of medals'}
)

fig_medals_per_sport.show()

Italian medals per Olympic games:

In [98]:
medals_per_games = (italydf_anon[italydf_anon['Medal'].notnull()]['Games'].value_counts().sort_index().reset_index())

medals_per_games.columns = ['Games', 'Count']

fig_medals_per_games = px.bar(
    medals_per_games,
    x='Games',
    y='Count',
    title="Italian medals per OS",
    labels={'Games': 'Olympic Games', 'Count': 'Number of medals'}
)

fig_medals_per_games.show()

Italian age distribution:

In [96]:
ita_age_min = italydf_anon['Age'].min()
ita_age_max = italydf_anon['Age'].max()

bins = list(range(int(ita_age_min - 5), int(ita_age_max + 5), 1))

fig_age_distribution_ita = px.histogram(
    italydf_anon,
    x='Age',
    nbins=len(bins),
    title="Age Distribution of Italian Athletes",
    labels={'Age': 'Age', 'count': 'Number of athletes'}
)

fig_age_distribution_ita.update_traces(marker_line_color="black", marker_line_width=1)
fig_age_distribution_ita.update_layout(bargap=0.05)

fig_age_distribution_ita.show()


# Equestrianism

Medal distribution between countries (NOC):

In [95]:
ita_df_equestrianism = italydf_anon[italydf_anon['Sport'] == 'Equestrianism']
df_equestrianism = df[df['Sport'] == 'Equestrianism']

medals_per_country = (df_equestrianism[df_equestrianism['Medal'].notnull()]['NOC'].value_counts().reset_index())
medals_per_country.columns = ['NOC', 'Count']

fig_eq_NOC_medal_distribution = px.bar(
    medals_per_country,
    x='NOC',
    y='Count',
    title="Equestrianism medals per country",
    labels={'NOC': 'Country', 'Count': 'Number of Medals'}
)

fig_eq_NOC_medal_distribution.show()


Age distribution:

In [88]:
equestrianism = italydf_anon[italydf_anon["Sport"] == "Equestrianism"].copy()
equestrianism["Group"] = "Equestrianism"

other_sports = italydf_anon[italydf_anon["Sport"] != "Equestrianism"].copy()
other_sports["Group"] = "Other sports"

age_compare = pd.concat([equestrianism, other_sports], ignore_index=True)

mean_age = (
    age_compare.groupby("Group")["Age"]
    .mean()
    .reset_index()
    .round(1)
)

fig_age_distribution_eq_vs_other = px.histogram(
    age_compare,
    x="Age",
    nbins=30,
    histnorm="percent",
    facet_row="Group",
    title="Age Distribution - Equestrianism vs Other Italian Sports",
    labels={"Age": "Age", "Group": "Group"}
)
fig_age_distribution_eq_vs_other.update_layout(
    height=700,
    margin=dict(t=80, b=40),
    font=dict(size=14),
)

# Adjust axes to remove the messy right-end overflow
fig_age_distribution_eq_vs_other.update_xaxes(range=[10, 50])     # adjust if you have older athletes
fig_age_distribution_eq_vs_other.update_yaxes(matches=None)
fig_age_distribution_eq_vs_other.show()

In [89]:
print(f"mean age equestrians: {ita_df_equestrianism['Age'].mean()}")
print(f"mean age non equestrians: {ita_df_otherSports['Age'].mean()}")

mean age equestrians: 32.47330960854092
mean age non equestrians: 25.692430278884462


Average activity span and age span for athletes:

In [90]:
age_span_per_person = ita_df_equestrianism.groupby('Name').agg(
    MinAge=('Age', 'min'),
    MaxAge=('Age', 'max'),
    ActiveYears=('Year', lambda x: x.nunique())
    ).reset_index()

average_min_age = age_span_per_person['MinAge'].mean()
average_max_age = age_span_per_person['MaxAge'].mean()
average_activity_span = age_span_per_person['ActiveYears'].mean()
longest_active = age_span_per_person['ActiveYears'].max()

average_min_age, average_max_age, average_activity_span

(np.float64(29.435294117647057),
 np.float64(33.576470588235296),
 np.float64(1.6956521739130435))

This calculation shows that on average and italian equestrian athlete competes between the ages of 29.4 and 33.6 with an average activity span of 1.7 Olympic Games.

In [91]:
bins = list(range(0, int(longest_active + 5), 1))

fig_eq_activity_years = px.histogram(
    age_span_per_person,
    x='ActiveYears',
    nbins=len(bins),
    title="Amount of Active Years for Italian Athletes",
    labels={"ActiveYears": "Number of Games", "count": "Numbr of Athletes"}
)

fig_eq_activity_years.update_traces(marker_line_width=1, marker_line_color="black")
fig_eq_activity_years.update_layout(bargap=0.05)

fig_eq_activity_years.show()

Gender distribution:

In [92]:
counts = ita_df_equestrianism.groupby(['Year','Sex']).size().unstack().fillna(0)
df_counts = counts.reset_index().melt(id_vars='Year', value_name='Count', var_name='Sex')

fig_eq_gender_distribution = px.bar(
    df_counts,
    x='Year',
    y='Count',
    color='Sex',
    barmode='group',
    title="Gender distribution of Italian equestrianism athletes over the years"
)

fig_eq_gender_distribution.show()

Italian Equestrianism medals per year:

In [105]:
all_years = ita_df_equestrianism['Year'].unique()
medals_per_year = ita_df_equestrianism[ita_df_equestrianism['Medal'].notnull()].groupby('Year').size()
medals_per_year = medals_per_year.reindex(all_years, fill_value=0).sort_index()

fig_eq_medals_per_year = px.bar(
    medals_per_year.reset_index(),
    x='Year',
    y=0,
    title="Italian Equestrianism medals per year",
    labels={'Year': 'Year', '0': 'Number of medals'}
)

fig_eq_medals_per_year.show()

In [ ]:
ita_eq_medals = ita_df_equestrianism.dropna(subset=['Medal']).drop_duplicates(subset=['Year', 'Medal', 'Event', 'ID'])
medals_type= ita_eq_medals.groupby(['Year', 'Medal']).size().reset_index(name="Count")
medals_type.head()

color_map = {
    "Gold": "#C6A907",
    "Silver": "#C0C0C0",
    "Bronze": "#CD7F32"
}

fig_eq_medals_type = px.bar(
    medals_type,
    x = 'Year',
    y = 'Count',
    color = 'Medal',
    color_discrete_map=color_map,
    title= "Medaljfördelning för Italien i Simning per OS-år",
    barmode='group'
)

fig_eq_medals_type.update_layout(
    xaxis_title = "OS År",
    yaxis_title = "Antal Medaljer"
)

fig_eq_medals_type.show()